### ISSUE TITLE
Reimplement filter options for Tadirah Technique and Object

### ISSUE THREAD

**Issue by patrickakk** [acdh-oeaw/dhcr-main#20]

**Comment by PixlTracer** [comment]
I looked through my emails and found an [entry in redmine](https://redmine.acdh.oeaw.ac.at/issues/18245?tab=history) which does not provide useful information
-- hence I went through the dhcr-main commits and found [this entry](https://github.com/acdh-oeaw/dhcr-main/commit/fc6bb70881fd34ddeba553a41eb18806389145db)- [this one](https://github.com/acdh-oeaw/dhcr-main/commit/2a66b85ea202ec9855d557ba72c07151847c1d64) and [this one](https://github.com/acdh-oeaw/dhcr-main/commit/9b913fcdcc2aeadfcff2bf75c938ddfe8d285d23) respectively, created by Hendrik, perhaps this could be of any use in resolving the issue?
you can find the commits from the timespan when the bug was reported [here](https://github.com/acdh-oeaw/dhcr-main/commits/prod?after=c6c25aa06f6fc4c87671e76de548eab97997f2fd+734&branch=prod&qualified_name=refs%2Fheads%2Fprod) (i.e. early March 2021).
I hope this helps!!

**Comment by patrickakk** [comment]
Thanks for the information. The Redmine issue Bug #18245 contains the following description: 

> When going to filters --> choose technique (any) and one wants to view the list, it says 'Error loading data'.
also, the 'filter' & 'sorting' buttons then disappear.

This issue was marked as 100% done, at 8 March 2021 on 18:40 by Hendrik.

I can see one commit which has impact on this. This is in another repo.
Repo: dhcr-core-plugin
Date: 8-3-2021
Commit: 07e675900d8c03fb98763f6fd29e4dd6f845278d
Link: https://github.com/acdh-oeaw/dhcr-core-plugin/commit/07e675900d8c03fb98763f6fd29e4dd6f845278d

This line is still the same, as of today. So nothing has been changed to this line, after this commit.

Do you know if this feature has been tested after 8-3-2021 and if it worked correctly?

**Comment by PixlTracer** [comment]
unfortunately I have no knowledge about this and whether it could be sorted out in the past.. perhaps digging deeper into the commit history could reveal some info? 
I've looked through [redmine](https://redmine.acdh.oeaw.ac.at/projects/course-registry) too, but there's not much included that could help on this matter (besides the tickets that I linked to in my previous entry)...

-- for the time being: would hiding the filters 'technique' and 'objects' be an option until the bug will be fixed?

**Comment by patrickakk** [comment]
Technically it would be possible to hide both filter buttons. This will remove this functionality from the user interface.

Is that the preferred solution? Please let me know what you want me to do.

**Comment by PixlTracer** [comment]
I'd be in favor of hiding the filter buttons for the time being.
and a side note: I've learned yesterday that php released a new version recently (27.10.), perhaps these changes are bug-related (/is tadirah technique/objects written in php)...?

**Comment by PixlTracer** [comment]
> Technically it would be possible to hide both filter buttons. This will remove this functionality from the user interface.
> 
> Is that the preferred solution? Please let me know what you want me to do.

I took the time to look into this bug briefly together with a dear colleague
-- the preferred solution would be to take time and solve the server side bug; 
Anyway, until then we should at least disable/hide the problematic filter functions until the bug is resolved and we can close this issue.

Additional info:
The bug can be re-produced, with e.g. using this URL 
https://dhcr.clarin-dariah.eu/api/v2/courses/index?tadirah_technique_id=8
which results in 
`{"message":"An Internal Error Has Occurred.","url":"\/courses\/index?tadirah_technique_id=8","error":{},"code":500}`

**Comment by patrickakk** [comment]
> the preferred solution would be to take time and solve the server side bug;
Anyway, until then we should at least disable/hide the problematic filter functions until the bug is resolved and we can close this issue.

Done. 

Both filter options are hidden in version 2022-11/2.

> and a side note: I've learned yesterday that php released a new version recently (27.10.), perhaps these changes are bug-related (/is tadirah technique/objects written in php)...?

A large part of the application is written in PHP. The bug was reported on 25-10-2022, which might exclude the changes in the new version.

**Comment by patrickakk** [comment]
> Additional info: The bug can be re-produced, with e.g. using this URL https://dhcr.clarin-dariah.eu/api/v2/courses/index?tadirah_technique_id=8 which results in `{"message":"An Internal Error Has Occurred.","url":"\/courses\/index?tadirah_technique_id=8","error":{},"code":500}`

Yes, it's a very consistent bug. ;)

I've made the filter buttons visible in the dev instance again, so we can check it there.

Both the API as well as the web application (dhcr-main) use the model for Courses which is defined in dhcr-core (mentioned above).

The error message from the logs of dhcr-main is:

> Error: [InvalidArgumentException] The `CoursesTadirahObjects` association is not defined on `Courses`.
> Valid associations are: Users, DeletionReasons, Countries, Cities, Institutions, CourseParentTypes, CourseTypes, Languages, CourseDurationUnits, Disciplines, TadirahTechniques, TadirahObjects in /workspace/vendor/cakephp/cakephp/src/ORM/Table.php on line 906


There are three many-to-many relations defined. One of them is working, this is Disciplines.
Two others don't work, this are: TadirahTechniques and TadirahObjects.

This is strange, since it looks like they are all defined in the same way in dhcr-core.

It's also not clear how long this bug exists and if it has worked before at all.

Maybe we want to talk in the next meeting about how much time should be spend on investigating and solving this?

**Comment by PixlTracer** [comment]
Yes, please write this into the agenda (that I laid my eyes on very briefly today - thanks for your input!!)

**Comment by patrickakk** [comment]
As discussed in the JFX of 30-11 this is put on hold.

**Comment by patrickakk** [comment]
@IvdL22 @PixlTracer 

The possibility has been added to filter on both Tadirah Techniques and Objects.

Can you review this and change the label to Done, if satisfied?

<img width="537" height="635" alt="Image" src="https://github.com/user-attachments/assets/5844cf20-d71d-40cc-b76c-87130eabcaa1" />

<img width="860" height="464" alt="Image" src="https://github.com/user-attachments/assets/dfd19d8c-94f2-443c-a21a-59ba297dc9a4" />

**Comment by vronk** [comment]
This is great. Thanks. can be released.

**Comment by patrickakk** [comment]
Implemented in 2025-08
